mbed_greentea/mbed_greentea_cli.py

```diff
@@ -782,8 +782,14 @@ def get_parallel_value(value):
                 # We will force configuration specific baudrate by adding baudrate to serial port
                 # Only add baudrate decoration for serial port if it's not already there
                 # Format used by mbedhtrun: 'serial_port' = '<serial_port_name>:<baudrate>'
-                if not mbed_dev['serial_port'].endswith(str(baudrate)):
-                    mbed_dev['serial_port'] = "%s:%d" % (mbed_dev['serial_port'], baudrate)
+                sp = mbed_dev['serial_port']
+                if sp:
```

Comment:
Instead can we add this check to the check at line https://github.com/ARMmbed/greentea/pull/201/files#diff-dede25ea3355a799ef1ec00a411ffd73R778

Here it gives the impression that checking serial port is related to annotating serial port with baud rate. But then in else we have ```continue```.